src/components/ProjectLineChart.tsx

```diff
@@ -0,0 +1,447 @@
+  LineChart,
+  Line,
+  XAxis,
+  YAxis,
+  Tooltip,
+  CartesianGrid,
+  Legend,
+  ResponsiveContainer,
+} from "recharts";
+import { useMemo, useState, useEffect } from "react";
+import { Card, CardContent } from "@/components/ui/card";
+import { Badge } from "@/components/ui/badge";
+import { BarChart3, ChevronLeft, ChevronRight } from "lucide-react";
+import { ChartDataPoint } from "@/lib/utils/chart";
+
+type Props = {
+  data: ChartDataPoint[];
+  weekMeta?: Record<number, { start_date: string; end_date: string }>;
+};
+
+type TooltipPayload = {
+  name: string;
+  value: number;
+  color: string;
+};
+
+export default function ProjectLineChart({ data, weekMeta }: Props) {
+  const [hoveredProject, setHoveredProject] = useState<string | null>(null);
+
+  const allProjectNames = useMemo(() => {
+    const projectNamesSet = new Set<string>();
+    data.forEach(dataPoint => {
+      Object.keys(dataPoint).forEach(key => {
+        if (key !== "week") {
+          projectNamesSet.add(key);
+        }
+      });
+    });
+    return Array.from(projectNamesSet);
+  }, [data]);
+
+  // Pagination state for weeks
+  const WINDOW_SIZE = 3;
+  const [windowStart, setWindowStart] = useState(0);
+  const maxWindowStart = Math.max(0, data.length - WINDOW_SIZE);
+  const visibleData = data.slice(windowStart, windowStart + WINDOW_SIZE);
+
+  // Project visibility state
+  const [visibleProjects, setVisibleProjects] = useState<
+    Record<string, boolean>
+  >(() => Object.fromEntries(allProjectNames.map((name) => [name, true])));
+
+  // Update visibleProjects if project list changes
+  useEffect(() => {
+    setVisibleProjects((prev) => {
+      const next: Record<string, boolean> = {};
+      for (const name of allProjectNames) {
+        next[name] = prev[name] !== undefined ? prev[name] : true;
+      }
+      return next;
+    });
+  }, [allProjectNames]);
+
+  const totalHoursPerWeek = useMemo(() => {
+    return data.map((d) => {
+      const total = allProjectNames.reduce(
+        (sum, project) => sum + (Number(d[project]) || 0),
+        0
+      );
+      return { week: d.week, total };
+    });
+  }, [data, allProjectNames]);
+
+  const projectColors = useMemo(() => {
+    const colors = [
+      "#0ea5e9", "#f59e0b", "#10b981", "#8b5cf6", "#f97316", "#06b6d4",
+      "#84cc16", "#ec4899", "#ef4444", "#6366f1", "#14b8a6", "#a855f7",
+      "#22c55e", "#3b82f6", "#fb7185",
+    ];
+    const mapping: Record<string, string> = {};
+    allProjectNames.forEach((project, idx) => {
+      mapping[project] = colors[idx % colors.length];
+    });
+    return mapping;
+  }, [allProjectNames]);
+
+  function getProjectColor(projectName: string) {
+    return projectColors[projectName] || "#000000";
+  }
+
+  return (
+    <Card className="w-full shadow-lg border-primary/10 bg-gradient-to-br from-white to-blue-50/30 dark:from-gray-900 dark:to-blue-950/30">
+      <div>
+        <h1 className="text-3xl font-bold mb-2">Weekly Project Hours</h1>
```

Comment:
![Screenshot 2025-06-20 at 14 13 41](https://github.com/user-attachments/assets/39a349cb-a1d5-4e68-b93b-970cb6d721e3)
Could you please add some left padding to this title?